cri/v1alpha2/cri.go

```diff
@@ -303,7 +282,42 @@ func (c *CriManager) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 // and we should reconfigure it with network plugin which will make sure it reacquire its original network configuration,
 // like IP address.
 func (c *CriManager) StartPodSandbox(ctx context.Context, r *runtime.StartPodSandboxRequest) (*runtime.StartPodSandboxResponse, error) {
-	// TODO Complete the details of function.
+
+	podSandboxID := r.GetPodSandboxId()
+
+	// start Podsandbox.
+	err := c.ContainerMgr.Start(ctx, podSandboxID, &apitypes.ContainerStartOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to start podSandbox %q: %v", podSandboxID, err)
+	}
+
+	// get the sandbox's meta data.
+	res, err := c.SandboxStore.Get(podSandboxID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get metadata of %q from SandboxStore: %v", podSandboxID, err)
+	}
+	sandboxMeta := res.(*SandboxMeta)
+
+	// Setup networking for the sandbox.
+	netnsPath, setupNetErr := c.setupPodNetwork(ctx, podSandboxID, sandboxMeta.Config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetNetwork(), sandboxMeta.Config)
+	if setupNetErr != nil {
+		return nil, setupNetErr
+	}
+	defer func() {
```

Comment:
If `setupNetErr` is not nil, the defer function will never being invoked.

Reply:
Yes, we should stop the container only if the network setting returns an error.